yast/lib/suse/connect/yast.rb

```diff
@@ -0,0 +1,260 @@
+        #   where a migration path is an array of OpenStruct objects with the attributes
+        #   identifier, arch, version, and release_type
+        def system_migrations(products, client_params = {})
+          _set_verify_callback(client_params[:verify_callback])
+          jsn_params = JSON.generate(client_params)
+          jsn_products = JSON.generate(products.map(&:to_h))
+          _process_result(GoConnect.system_migrations(jsn_params, jsn_products))
+        end
+
+        # Lists all available offline migration paths for a given list of products.
+        # Accepts an array of products, and returns an array of possible
+        # migration paths. A migration path is a list of products that may
+        # be upgraded.
+        #
+        # @param installed_products [Array <OpenStruct>] the list of currently installed {Product}s in the system
+        # @param target_base_product [OpenStruct] the {Product} that the system wants to upgrade to
+        # @param client_params [Hash] parameters to instantiate {Client}
+        #
+        # @return [Array <Array <OpenStruct>>] the list of possible migration paths for the given {Product}s,
+        #   where a migration path is an array of OpenStruct objects with the attributes
+        #   identifier, arch, version, and release_type
+        def system_offline_migrations(installed_products, target_base_product, client_params = {})
+          _set_verify_callback(client_params[:verify_callback])
+          jsn_params = JSON.generate(client_params)
+          jsn_products = JSON.generate(installed_products.map(&:to_h))
+          jsn_target = JSON.generate(target_base_product.to_h)
+          _process_result(GoConnect.offline_system_migrations(jsn_params, jsn_products, jsn_target))
+        end
+
+        # List available Installer-Updates repositories for the given product
+        #
+        # @param [Remote::Product] list repositories for this product
+        # @param [Hash] client_params parameters to instantiate {Client}
+        #
+        # @return [Array <OpenStruct>] list of Installer-Updates repositories
+        def list_installer_updates(product, client_params = {})
+          _set_verify_callback(client_params[:verify_callback])
+          jsn_params = JSON.generate(client_params)
+          jsn_product = JSON.generate(product.to_h)
+          _process_result(GoConnect.list_installer_updates(jsn_params, jsn_product))
+        end
+
+        # Writes the config file with the given parameters, overwriting any existing contents
+        # Only persistent connection parameters (url, insecure) are written by this method
+        # Regcode, language, debug etc are not
+        # @param [Hash] client_params
+        #  - :insecure [Boolean]
+        #  - :url [String]
+        def write_config(client_params = {})
+          _set_verify_callback(client_params[:verify_callback])
```

Comment:
does this really need the `_set_verify_callback` ?

Reply:
not really.